### ISSUE TITLE
False positive DCN_NULLPOINTER_EXCEPTION around Collection.contains()

### ISSUE THREAD

**Issue by rovarga** [spotbugs/spotbugs#1954]
The java.util.Collection.contains() method documents NPE (and CCE) as being legally thrown. Hence defensive code inquiring about presence needs to look like this:
```  
  List<?> list; // known to be non-null, but unknown implementation
  Object obj; // nullable
  boolean found;
  try {
    found = list.contains(obj);
  } catch (ClassCastException | NullPointerException e) {
    found = false;
  }
```

This elicits DCN_NULLPOINTER_EXCEPTION. SpotBugs should recognize when the NPE catch block contains only a method call to Collection.contain() and not issue the warning. Similar methods are Collection.{add,remove}(), List.{indexOf,lastIndexOf}(), Map.contains{Key,Value}() and few others.

**Comment by rovarga** [comment]
Sure thing, I will comment on the page.

Nevertheless, a pre-existing comment already states that:
```By that argument, ERR08-J should not be a hard-and-fast rule, but rather a recommendation.```
